src/app/encode/services/EncodeUserService.ts

```diff
@@ -94,6 +95,12 @@ export class EncodeUserService {
         
         await this._dbService.updateEncodeUser(this._user);
     }
+
+    public async abandonTest() {
```

Comment:
Cambiar por:

```suggestion
    public abandonTest(): Promise<void> {
        this._user.abandonedByUser = true;
        return this.updateUserInDB();
    }
```

fijate que va sin el navigate.